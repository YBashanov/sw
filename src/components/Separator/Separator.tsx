import React from 'react';
import { useIntl } from 'react-intl';

import './index.less';

type SeparatorType = {
    label?: React.ReactNode;
};

/**
 * Разделитель
 */
const Separator: React.FC<SeparatorType> = props => {
    const { formatMessage: f } = useIntl();
    const { label = f({ id: 'App.Separator.DefaultLabel' }) } = props;

    return (
        <div className="separator">
            <div className="label">{label}</div>
        </div>
    );
};

export default Separator;
