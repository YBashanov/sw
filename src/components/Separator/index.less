.separator {
  display: block;
  border-bottom: 1px solid #bbbbbb;

  .label {
    font-size: 14px;
    display: inline-block;
    position: relative;
    top: 8px;
    left: 20px;
    padding: 0 8px;
    background-color: white;
  }
}
