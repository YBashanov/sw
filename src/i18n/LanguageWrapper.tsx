import React, { createContext, useState } from 'react';
import { getIntlLocale, getIntlMessages } from '@/i18n';
import { IntlProvider } from 'react-intl';

export type Language = 'ru' | 'en';

export const defaultLanguage: Language = 'ru';

type LanguageContextType = {
    language?: Language;
    setLanguage?: (language: Language) => void;
};

type LanguageWrapperType = {
    children: React.ReactNode;
};

export const LanguageContext = createContext<LanguageContextType>({
    language: defaultLanguage,
});

export const defaultSaveLang = localStorage.getItem('language') as Language;

const LanguageWrapper: React.FC<LanguageWrapperType> = ({ children }) => {
    const [language, setLanguage] = useState<Language>(defaultSaveLang || defaultLanguage);

    const handleSetLanguage = (lang: Language) => {
        localStorage.setItem('language', lang);

        // установка для React-компонентов
        setLanguage(lang);

        // установка для статики (схемы Yup, модели Redux)
        // устанавливается в i18n/index

        // требуется перезагрузка для сохранения в localStorage и взятия оттуда
        location.href = '/';
    };

    // тестирование смены языка. (в идеале это должно быть по кнопке)
    React.useEffect(() => {
        // const wantLang = 'ru';
        // if (defaultSaveLang !== wantLang) handleSetLanguage(wantLang);
    }, []);

    return (
        <LanguageContext.Provider value={{ language, setLanguage: handleSetLanguage }}>
            <IntlProvider locale={getIntlLocale(language)} messages={getIntlMessages(language)}>
                {children}
            </IntlProvider>
        </LanguageContext.Provider>
    );
};

export default LanguageWrapper;
