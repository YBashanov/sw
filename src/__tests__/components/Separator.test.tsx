import React from 'react';
import { describe, expect, it } from '@jest/globals';
import { render } from 'enzyme';
import Separator from '@/components/Separator/Separator';
import LanguageWrapper from '@/i18n/LanguageWrapper';

describe('testing: Separator (component)', function () {
    const JSX = (
        <LanguageWrapper>
            <Separator />
        </LanguageWrapper>
    );
    const Component = render(JSX);

    it('Snapshot correctly', function () {
        expect(Component).toMatchSnapshot();
    });

    it('Static HTML', function () {
        expect(Component.text()).toEqual('Дополнительно');
    });
});
