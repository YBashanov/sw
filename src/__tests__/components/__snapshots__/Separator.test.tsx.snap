// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing: Separator (component) Snapshot correctly 1`] = `
LoadedCheerio {
  "0": <div
    class="separator"
  >
    <div
      class="label"
    >
      Дополнительно
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
